src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs

```diff
@@ -661,37 +657,33 @@ internal abstract partial class SafeDeleteContext : SafeHandle
             SafeFreeContextBuffer outFreeContextBuffer = null;
             Span<Interop.SspiCli.SecBuffer> outUnmanagedBuffer = stackalloc Interop.SspiCli.SecBuffer[2];
             outUnmanagedBuffer[1].pvBuffer = IntPtr.Zero;
+            Span<IntPtr> ptr = stackalloc IntPtr[3];
             try
             {
-                Span<Interop.SspiCli.SecBuffer> inUnmanagedBuffer = stackalloc Interop.SspiCli.SecBuffer[inSecurityBufferDescriptor.cBuffers];
+                Span<Interop.SspiCli.SecBuffer> inUnmanagedBuffer = stackalloc Interop.SspiCli.SecBuffer[inSecBuffers.Count];
```

Comment:
I believe the intention is for the stackalloc to be 1..3 entries long but there's no assertion or check that i can see and since count is unconditionally increased in SetNextBuffer it could in theory any value if someone found a way to abuse it.

Reply:
Can you share an example of where/when it wouldn't be a small value?

Reply:
Nothing obvious, as i said the intention is clearly that it should be 1..3 which causes no issue. I just thought it might be worth an assert or check.

Reply:
I see.  Sure, adding an assert would be fine.

Reply:
There is Assert in SetNextBuffer so it won't go more than 3. I can also make it consistent and always allocate 3 since the rest of the logic is fixed anyway. I did not find a good way how to make the equivalent of `fixed()` in a more flexible way.

Reply:
Not sure if hot path but having a constant-sized stack via `(stackalloc Interop.SspiCli.SecBuffer[3])[0..inSecBuffers.Count]` may use fewer instructions, if you do have this upper bound.

Reply:
Right, if the max is actually 3, just use 3.  The JIT is better at optimizing const-sized stackallocs.  There's also no reason to slice it beyond that unless you actually use the length later.  If you do use the length later, just stick with the variable-sized stackalloc.

Reply:
I updated code to use const-sized allocation since the difference between 0 and 3 is really small. (and will be 2 in most cases) 
I was wondering if '3' should be some named constant instead of magic number but I not sure what the name should be and where it should live.